inference-engine/src/mkldnn_plugin/emitters/jit_snippets_emitters.hpp

```diff
@@ -0,0 +1,641 @@
+                    h->add(pointer, reg_tmp_64);
+                }
+            }
+        };
+        for (auto i = 0; i < num_params; i++) {
+            regs[i] = Reg64(reg64_tmp_start + i);
+            if (i < num_inputs)
+                h->mov(regs[i], h->ptr[reg_const_params + GET_OFF(src_ptrs) + i * sizeof(void*)]);
+            else
+                h->mov(regs[i], h->ptr[reg_const_params + GET_OFF(dst_ptrs) + (i - num_inputs) * sizeof(void*)]);
+            init_ptrs_with_offsets(regs[i], &jcp.data_offsets[i * harness_num_dims]);
+        }
+
+        for (auto& c : code) {
+            c.first->emit_code(c.second.first, c.second.second, pool, gpr);
+        }
+
+        h->postamble();
+    }
+
+    std::vector<std::pair<std::shared_ptr<Emitter>, ngraph::snippets::RegInfo>> code;
+};
+///
+/// \brief    Tile is designed to organize loop over the input and output data. It is essentially a for(...) loop:
+/// it calculates the total number of iterations, performs operations specified by enclosed emitters, advances iteration counters
+/// and breaks when necessary.
+///
+/// \param      inc           The number of input entities (or scheduler counts) processed during one iteration of the tile.
+/// It is expected to be 1 for outer or scalar tiles and vlen for vector tiles.
+/// \param      previous_inc  Increment of the previous Tile in current dimension. Must be 0 if this is the first Tile.
+/// So previous_inc is zero for outer and vector tiles (the are the first in dim) and vlen for scalar tiles (they usually go after vector Tiles).
+/// \param      num_params    sum number inputs and number of outputs of the node.
+/// \param      dim           dimension of the tile. Note that only 2d Tile are currently supported, so dim is 0 for outer tiles, 1 for inner tiles.
+///
+// Todo: Inner and outer tiles have different semantics. For example, outer tile always has the increment == 1, and it can contain only
+//  tile emitters (one outer or two inner). So it seems better to create different classes for inner and outer tiles.
+// Todo: Currently data pointers incremented after each read/write in Load/Store emitters, so we have to decrement them here
+//  if the same data needs to be read twice. Better to move all the pointer increments to TileEmitter and avoid the increments if necessary.
+class TileEmitter : public jit_emitter {
+public:
+    TileEmitter(mkldnn::impl::cpu::x64::jit_generator* h, mkldnn::impl::cpu::x64::cpu_isa_t isa,
+    const std::shared_ptr<ov::Node>& n)
+    : jit_emitter(h, isa, n) {
+        const auto tile = ov::as_type_ptr<ngraph::snippets::op::Tile>(n);
+        if (!tile)
+            IE_THROW() << "TileEmitter invoked with invalid op argument";
+        if (!tile->compile_params)
+            IE_THROW() << "TileEmitter invoked without compile_params";
+        code = tile->region;
+        jcp = *reinterpret_cast<const jit_snippets_compile_args*>(tile->compile_params);
```

Comment:
Shouldn't we check on nullptr?

Reply:
Already checking: if (!tile->compile_params)